libvast/src/partition_synopsis.cpp

```diff
@@ -139,4 +139,33 @@ unpack(const fbs::partition_synopsis::v0& x, partition_synopsis& ps) {
   return caf::none;
 }
 
+// This overload is for supporting the transition from VAST 2021.07.29
+// to 2021.08.26, where the `id_range` field was added to partition
+// synopsis. It can be removed once these versions are unsupported.
+caf::error unpack(const fbs::partition_synopsis::v0& x, partition_synopsis& ps,
+                  uint64_t offset, uint64_t events) {
```

Comment:
Can we express this in terms of the other (or vice versa) to avoid redundancy? This is a lot of duplicated code currently.

Reply:
I extracted the loop into a separate function.

Also, one nice thing is that this function does not need to be kept in sync, as it will only be used for partition synopses created before now. So any future features only need to be added to the regular `unpack()` overload.